marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs

```diff
@@ -70,7 +70,8 @@ function main(contentSequence, options, stepExecutionContext) {
       }
 
       let doc = content.value;
-      let instance = lib.extractInstance(doc);
+      const sourceRecordInstanceOnly = mappingStep.sourceRecordScope == "entireRecord" ? false : true;
```

Comment:
This looks like a function waiting to be extracted, as the logic is duplicated above in lib.sjs . And then ideally, the backend endpoint that retrieves a source doc would use the same function to either return the full doc or just the instance portion.